tensorflow/core/kernels/matmul_op.cc

```diff
@@ -591,6 +591,19 @@ TF_CALL_bfloat16(REGISTER_CPU);
 TF_CALL_int32(REGISTER_CPU);
 TF_CALL_complex64(REGISTER_CPU_EIGEN);
 TF_CALL_complex128(REGISTER_CPU_EIGEN);
+
+#elif defined(INTEL_MKL) && defined(DO_NOT_USE_ML)
+
+// MKL opensource supports only float for matrix-multiplication
+TF_CALL_float(REGISTER_CPU_EIGEN);
```

Comment:
Why do we register with the "eigen" label to call MKL instead of Eigen?  Confusing...